logtail/logtail.go

```diff
@@ -287,26 +296,31 @@ func (l *logger) uploading(ctx context.Context) {
 	}
 }
 
-func (l *logger) upload(ctx context.Context, body []byte) (uploaded bool, err error) {
+// upload uploads body to the log server.
+// origlen indicates the pre-compression body length.
+// If origlen is -1, the body is not compressed.
+func (l *logger) upload(ctx context.Context, body []byte, origlen int) (uploaded bool, err error) {
 	req, err := http.NewRequest("POST", l.url, bytes.NewReader(body))
 	if err != nil {
 		// I know of no conditions under which this could fail.
 		// Report it very loudly.
 		// TODO record logs to disk
 		panic("logtail: cannot build http request: " + err.Error())
 	}
-	if l.zstdEncoder != nil {
+	if origlen != -1 {
```

Comment:
this doesn't seem to match the docs.

Reply:
Why?

`// If origlen is -1, the body is not compressed.`

The following lines of code (setting "zstd", writing original length) only make sense to do when the body is compressed. So we need to change origlen != -1.

But maybe I'm missing something. I tweaked the comments to perhaps be clearer about who is doing the compression.